### PR TITLE
Add GPT2ForTokenClassification

### PR DESCRIPTION
**Pull request by tucan9389**
# What does this PR do?

- Add `GPT2ForTokenClassification` class for GPT2 upstream and NER downstream task

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@patrickvonplaten, @LysandreJik
@sgugger, @patil-suraj

### CODE REVIEW

**Comment by tucan9389** [comment]
Fixed all test failures and all passed.

File: src/transformers/modeling_outputs.py [hunk 1]
```diff
@@ -449,6 +449,42 @@ class SequenceClassifierOutputWithPast(ModelOutput):
     attentions: Optional[Tuple[torch.FloatTensor]] = None
 
 
+@dataclass
+class TokenClassifierOutputWithPast(ModelOutput):
```
[the remark this answers is not in the document]
Comment (tucan9389) [reply src/transformers/modeling_outputs.py:453]: @patrickvonplaten
Thanks for your comment. 
Oh, I agree with that. I'll change to `TokenClassifierOutput`.

**Review by sgugger — APPROVED** [review]
Thanks a lot for adding this model!

**Comment by tucan9389** [comment]
@sgugger Thanks for approving.

@patrickvonplaten  I just fixed some errors that occurred in `run_tests_torch` CI test and all tests were passed.

**Review by patrickvonplaten — APPROVED** [review]
Very nice! Thanks for you work here @tucan9389

**Pull request merged**